pkg/cmd/server/origin/master_config.go

```diff
@@ -673,6 +674,21 @@ func newAuthenticator(config configapi.MasterConfig, restOptionsGetter restoptio
 		},
 	}
 
+	//	if we have a front proxy providing authentication configuration, wire it up and it should come first
+	if config.AuthConfig.RequestHeader != nil {
+		requestHeaderAuthenticator, err := headerrequest.NewSecure(
+			config.AuthConfig.RequestHeader.ClientCA,
+			config.AuthConfig.RequestHeader.ClientCommonNames,
+			config.AuthConfig.RequestHeader.UsernameHeaders,
```

Comment:
I do not think we do so in integration tests (which is why I had to specify it directly).